README.md

```diff
@@ -1,53 +1,144 @@
+
+```
+Post.all
+```
+
+If everything is setup properly you will see that it returns an empty ActiveRecord object. Now let's test creating a record using the console:
+
+```
+Post.create!(title: "My title", description: "The post description")
+```
+Now running the query:
+
+```
+Post.last
+```
+
+Returns our newly created post.
+
+With our Post model working, let's add a new feature that returns a summary of a post, as usual start off by creating a spec for the feature:
+
+```ruby
+it 'has a summary' do
+  post = Post.create!(title: "My title", description: "The post description")
+  expect(post.post_summary).to eq("My title - The post description")
+end
+```
+
+If we run this we'll get a failure since we do not have a ```post_summary``` method for post. Add that to the model file:
+
+```ruby
+def post_summary
+end
+```
+This now results in a failure since the method currently doesn't return anything. Update the ```post_summary``` method like below:
+
+```ruby
+def post_summary
+  self.title + " - " + self.description
+end
+```
+Now if you run the tests, all of them are passing and our Post model has an instance method that returns a post summary. You can test this out in the Rails console as well by running a query on the record we created, such as:
+
+```
+Post.last.post_summary
+```
+And it will return the summary value of that specfic post.
+
+As you may have noticed, we did not have to create a controller, route, view, etc. in order to get the Post model working. The data aspect of the application can work separately from the view and data flow logic, this level of abstraction makes it straightforward to test data behavior without having it strongly coupled to how it is rendered to the user. With that being said, it is considered a best practice to have your controller and view files follow the proper naming convention so that the MVC associations are straightforward to follow. For example, to build out the controller and view code for our Post model we would create the following structure:
+
+* Create a ```posts_controller.rb``` file that calls on the Post model
+* Create a ```views/posts/``` directory that stores the views related to the Post model
+
+Also, if you are coming from other programming languages, you may be wondering how exactly we are able to connect to the database automatically, without having to create connection strings. The reason for this simplicity resides in the ```config/database.yml``` file that was generated when we created our application and ran ```rake db:create```, in that file you will see that the development, test, and production databases are all configured. From that stage the ActiveRecord::Base.connection method connects your application to the database, which is another benefit from having our model classes inherit from the ActiveRecord::Base module.
```

Comment:
This is valuable stuff, but a bit too much and out of context for this reading. Rails environment stuff is important, but we are saving it for later units